src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -128,9 +133,9 @@
 
     private Map <String,List <String>> vanityTargets;
 
-    private volatile Map<String, Map<String, String>> aliasMap;
+    private Map<String, Map<String, String>> aliasMap;
     
-	private volatile boolean isAliasMapInitialized = false;
+	private boolean isAliasMapInitialized = false;
```

Comment:
Sorry for the late reply, I missed your additional changes.

Right now the situation is no better than in the original submission. You're reading/writing to `aliasMap` and `isAliasMapInitialized` from different threads and there is no guarantee that the changes will be visible - no synchronized/volatile/etc.

The aliasMap may be a concurrent implementation, but the reference setting is not thread-safe. The solution you choose should at least:

- ensure that aliasMap/isAliasMapInitialized values are changed atomically
- access to aliasMap/isAliasMapInitialized is done in a thread-safe manner

Reply:
@rombert  ok I revisited the scenario and added the volatile markers again. This should now be sufficient as access is now threadsafe - the aliasmap is not manipulated in the thread despite of assigning a new map I don't see another multi-threading issue.

Reply:
The changes you made don't ensure that the values are changed atomically. Assuming that we have thread `T1` updating the values and thread `T2` reading the values it's possible for thread `T1` to set value 1, then `T2` to read value 1 (new) and value 2 (old).

You should ensure that the values are changed atomically, for instance by using synchronized blocks or by using a single volatile field that holds both the aliasMap data and the init ready flag.